src/main/java/subway/application/LineStationService.java

```diff
@@ -0,0 +1,79 @@
+package subway.application;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.controller.dto.request.AddInitStationToLineRequest;
+import subway.controller.dto.request.AddStationToBetweenLineRequest;
+import subway.controller.dto.request.AddStationToEndLineRequest;
+import subway.controller.dto.request.RemoveStationOnLineRequest;
+import subway.domain.Line;
+import subway.domain.Station;
+import subway.domain.vo.Distance;
+import subway.exception.BusinessException;
+import subway.persistence.LineRepository;
+import subway.persistence.StationRepository;
+
+@Service
+@Transactional
+public class LineStationService {
+
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
```

Comment:
```suggestion
    private final LineService lineService;
    private final StationService stationService;
```
이렇게 Service를 DI하는 것과 Respository를 DI는 것을 비교해보고 장단점을 고민해보면 어떨까요??

Reply:
service를 service가 di 각 도메인의 service마다 해당 도메인의 repository를 의존하게 되니 repository의 공통적인 기능에 접근할 때, 동일한 처리를 해줄 수 있어서 조금 더 추상화가 되고 코드 중복을 줄일 수 있겠네요! 트랜잭션도 필요한 경우 정말 최소 단위로 쪼갤 수도 있을 것 같습니다. 하지만 그렇게되면 service가 어쩔 수 없이 domain을 반환하게 되고 controller에서 service에 접근이 가능하니 controller에서 service를 통해 domain을 알게되는 경우가 생길 것 같습니다. 제 생각에는 장점보다 단점이 큰 것 같은데 시카의 생각은 어떤가요??

Reply:
장점에 대해서 정리를 잘해주셨네요...! 👏👏

단점에 대해서는, 단순히 "controller에서 domain을 알게된다"라는 것이 정확히 장점에 비해 어떤점 더 큰 단점인지 설명해주실 수 있을까요??
단순히 알게된다라는 것 자체가 문제라면 계층과 격리를 위한 DTO를 사용하면 되는 것 아닐까요?